File: pkg/codegen/testing/test/testdata/single-or-none-pp/nodejs/single-or-none.ts
import * as pulumi from "@pulumi/pulumi";

function singleOrNone<T>(elements: pulumi.Input<T>[]): pulumi.Input<T> {
    if (elements.length != 1) {
        throw new Error("singleOrNone expected input list to have a single element");
    }
    return elements[0];
}

export const result = singleOrNone([1]);

File: pkg/codegen/testing/utils/testdata/single-or-none-pp/go/single-or-none.go
package main

import (
	"fmt"

	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func singleOrNone[T any](elements []T) T {
	if len(elements) != 1 {
		panic(fmt.Errorf("singleOrNone expected input slice to have a single element"))
	}
	return elements[0]
}

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		ctx.Export("result", pulumi.Float64(singleOrNone([]float64{
			1,
		})))
		return nil
	})
}
